sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/IndentSingleLineFollowingConditional.cs

```diff
@@ -0,0 +1,185 @@
+                CheckDo, SyntaxKind.DoStatement);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckFor, SyntaxKind.ForStatement);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckForEach, SyntaxKind.ForEachStatement);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckIf, SyntaxKind.IfStatement);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckElse, SyntaxKind.ElseClause);
+        }
+
+        private static void CheckWhile(SyntaxNodeAnalysisContext context)
+        {
+            var whileStatement = (WhileStatementSyntax)context.Node;
+            if (!IsStatementIndentationOk(whileStatement, whileStatement.Statement))
+            {
+                // Squiggle - "while (condition1 && condition2)"
+                var primaryLocation = GetLocationBetweenTokens(whileStatement.WhileKeyword, whileStatement.CloseParenToken);
+               
+                ReportIssue(context, primaryLocation, whileStatement.Statement, "while");
+            }
+        }
+
+        private static void CheckDo(SyntaxNodeAnalysisContext context)
+        {
+            var doStatement = (DoStatementSyntax)context.Node;
+            if (!IsStatementIndentationOk(doStatement, doStatement.Statement))
+            {
+                // Just highlight the "do" keyword
+                ReportIssue(context, doStatement.DoKeyword.GetLocation(), doStatement.Statement, "do");
+            }
+        }
+
+        private static void CheckFor(SyntaxNodeAnalysisContext context)
+        {
+            var forStatement = (ForStatementSyntax)context.Node;
+            if (!IsStatementIndentationOk(forStatement, forStatement.Statement))
+            {
+                // Squiggle - "for (...)"
+                var primaryLocation = GetLocationBetweenTokens(forStatement.ForKeyword, forStatement.CloseParenToken);
+
+                ReportIssue(context, primaryLocation, forStatement.Statement, "for");
+            }
+        }
+
+        private static void CheckForEach(SyntaxNodeAnalysisContext context)
+        {
+            var forEachStatement = (ForEachStatementSyntax)context.Node;
+            if (!IsStatementIndentationOk(forEachStatement, forEachStatement.Statement))
+            {
+                // Squiggle - "foreach (...)"
+                var primaryLocation = GetLocationBetweenTokens(forEachStatement.ForEachKeyword, forEachStatement.CloseParenToken);
+
+                ReportIssue(context, primaryLocation, forEachStatement.Statement, "foreach");
+            }
+        }
+
+        private static void CheckIf(SyntaxNodeAnalysisContext context)
+        {
+            IfStatementSyntax ifStatement = (IfStatementSyntax)context.Node;
+
+            // Special case for "else if" on the same line.
+            // In that case, we'll check that the statement is more indented then the "else", not the "if".
```

Comment:
This special case was highlighted by ITs - it's used frequently in Nancy and Ember.
Also, it's the default indentation used by VS if you use _else if_ then hit _Enter_.